### PR TITLE
add project euler for problem 50

### PR DESCRIPTION
**Pull request by willtryagain**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_050/sol1.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    is_prime_number = [True] * (limit + 1)
+    i = 2 # begin from the smallest prime
+    while i * i <= limit:
+        if is_prime_number[i]:
+            for j in range(2*i, limit + 1, i):
+                is_prime_number[j] = False
+        i += 1
+    return is_prime_number
+
+def solution(limit: int = 1_000_000):
```
Comment (xcodz-dot) [inline project_euler/problem_050/sol1.py:48]:
```suggestion
def solution(limit: int = 1_000_000) -> int:
```
type hinting for return values

File: project_euler/problem_050/sol1.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    # factor less than equal to its square root
+    
+    while i * i <= number:
+        # if number is divisible by i then i is a factor of number
+        if  number % i == 0:
+            return False
+        i += 1
+    return True
+
+def sieve_of_eratosthenes(limit: int) -> [list]:
```
Comment (xcodz-dot) [inline project_euler/problem_050/sol1.py:34]:
```suggestion
def sieve_of_eratosthenes(limit: int) -> list:
```
brackets are not needed

File: project_euler/problem_050/sol1.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+The longest sum of consecutive primes below one-thousand that adds to a
+ prime, contains 21 terms, and is equal to 953.
+
+Which prime, below one-million, can be written as the sum of the 
+most consecutive primes?
+"""
+
+
+def is_prime(number: int) -> [bool]:
```
Comment (xcodz-dot) [inline project_euler/problem_050/sol1.py:18]:
```suggestion
def is_prime(number: int) -> bool:
```
brackets are not required

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
required changes to type hinting of return value

File: project_euler/problem_050/sol1.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                break
+            if is_prime(total):
+                if max_length < j - i + 1:
+                    prime_number = total
+                    max_length = j - i + 1
+        
+    return prime_number
+
+if __name__ == "__main__":
+    print(f"{solution() = }")
```
Comment (xcodz-dot) [inline project_euler/problem_050/sol1.py:79]:
```suggestion
    print(f"{solution() = }")

```
PEP 8 fix required for action to complete

File: project_euler/problem_050/sol1.py (hunk 2 above)
Comment (xcodz-dot) [reply project_euler/problem_050/sol1.py:34]: @willtryagain this is also needed to be accepted

File: project_euler/problem_050/sol1.py (hunk 4 above)
Comment (xcodz-dot) [reply project_euler/problem_050/sol1.py:79]: @willtryagain, this will fix a github workflow error

File: project_euler/problem_050/sol1.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+
+
+
```
Comment (xcodz-dot) [inline project_euler/problem_050/sol1.py:78]:
```suggestion

```
These are too many new lines

File: project_euler/problem_050/sol1.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                    prime_number = total
+                    max_length = j - i + 1
+        
+    return prime_number
+
+if __name__ == "__main__":
+    print(f"{solution() = }")
+
+
+
```
Comment (xcodz-dot) [inline project_euler/problem_050/sol1.py:77]:
```suggestion
```

These are too many newlines

**Comment by Takosaga** [comment]
You have the same pre-commit hooks as I did, these helped me figure it out
https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut

https://www.geeksforgeeks.org/python-code-formatting-using-black/

**Comment by xcodz-dot** [comment]
> You have the same pre-commit hooks as I did, these helped me figure it out
> https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut
> 
> https://www.geeksforgeeks.org/python-code-formatting-using-black/

You are right @Takosaga, @willtryagain can use black, it will be good for his work to get done instantly.

**Comment by willtryagain** [comment]
> You have the same pre-commit hooks as I did, these helped me figure it out
> https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut
> 
> https://www.geeksforgeeks.org/python-code-formatting-using-black/

Thank you @Takosaga, now it does pass black. Can you tell 
How do we fix End of File?

**Comment by Takosaga** [comment]
@willtryagain oh that is a different hook than me, but seems like it is Fixing project_euler/problem_050/__init__.py. That file should be an empty file. It took me while to figure out this stuff but there is some more information in the .pre-commit-config.yaml for the hooks

**Comment by xcodz-dot** [comment]
@willtryagain, try using black on `__init__.py` file as well. It might fix it

File: project_euler/problem_050/__init__.py [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
Comment (xcodz-dot) [inline project_euler/problem_050/__init__.py:1]:
```suggestion


```
If black does not help than apply this change

**Comment by Takosaga** [comment]
This should be how your **__init__.py** file should be. That might be what is catching your hook
empty

File: project_euler/problem_050/__init__.py (hunk 7 above)
Comment (xcodz-dot) [inline project_euler/problem_050/__init__.py:1]:
```suggestion
```

Try deleting this line

**Review by xcodz-dot — CHANGES_REQUESTED** [review]

**Comment by xcodz-dot** [comment]
rename `__init__.py.txt` -> `__init__.py`

File: project_euler/problem_050/__init__.py [hunk 8]
```diff
@@ -0,0 +1 @@
...
```
Comment (xcodz-dot) [inline project_euler/problem_050/__init__.py:1]: Try making a completly empty `__init__.py` file with no content and not even a single new line

**Review by xcodz-dot — CHANGES_REQUESTED** [review]

**Comment by xcodz-dot** [comment]
I got the perfect solution:
* Download the repository
* Make sure you Have git installed
* run `pip install pre-commit`
* go to the repository directory and run `pre-commit install`
* also run `pre-commit run --all-files`

This is garuenteed to fix your issues

**Comment by dhruvmanila** [comment]
Thank you for your contribution but we already have another solution for it which was opened and got accepted. Maybe try solving another problem which is not present in our directory?

**Pull request closed without merging**